r1cs/gadgets/std/src/fields/fp12.rs

```diff
@@ -856,13 +952,19 @@ where
 }
 
 impl<P, ConstraintF: PrimeField + SquareRootField> ThreeBitCondNegLookupGadget<ConstraintF>
-    for Fp12Gadget<P, ConstraintF>
-where
-    P: Fp12Parameters,
-    <P::Fp6Params as Fp6Parameters>::Fp2Params: Fp2Parameters<Fp = ConstraintF>,
+for Fp12Gadget<P, ConstraintF>
+    where
+        P: Fp12Parameters,
+        <P::Fp6Params as Fp6Parameters>::Fp2Params: Fp2Parameters<Fp = ConstraintF>,
 {
     type TableConstant = Fp12<P>;
 
+    /* Three bit look up circuit combining a two bit table and conditional negation.
+    Outputs
+        c = (-1)^b[2] * c[i] = (-1)^b[2] * c[b[1]*2 + b[0]]
```

Comment:
Since that each implementation from Fp2Gadget above uses the underlying FpGadget implementation, there's no need to comment the same thing again. I would remove it here and in all the other files (except fp.rs)